cmd/tink-server/main.go

```diff
@@ -32,7 +33,14 @@ func main() {
 	errCh := make(chan error, 2)
 	facility := os.Getenv("FACILITY")
 
-	cert, modT := rpcServer.SetupGRPC(ctx, logger, facility, errCh)
+	tinkDB := db.Connect(logger)
+	numAppliedMigrations, err := tinkDB.Migrate()
+	if err != nil {
+		panic(err)
+	}
+	log.With("num_applied_migrations", numAppliedMigrations).Info("Running migrations if necessary")
```

Comment:
We've found out the hard way that its best to allow for migrations to run separately from normal service startup. This way once we have multiple tinks running an upgrade won't have +1 tinks trying to migrate the db.

Reply:
I didn't take this into consideration. Now the idea to migrate as part of the start is way less appealing :) I will close this PR in favor of something else :D because the title is kind of clear

Reply:
Well if you start up 2 tink-servers at the same time they may both try to do some costly migration and take up db time and only one will succeed, the other will fail and restart and then continue. It would stay correct, but waste time/db resources.

Having `--migrate-only` would be better imo, as it can be a separate step in the deployment pipeline and would exit when done.

Reply:
Do you think that flag should stay in tink-server? I like that! I will check if the library supports something like a dry run that checks for migrations but does not apply them. In this way I can run it at start and notify the user that there are migration not applied yet

Reply:
Done @mmlb and I have updated the PR description with the new workflow

Reply:
yeah the dry run would be pretty nice imo.